### DISCUSSION TITLE
How to show the callback result in the message?

### DISCUSSION THREAD

**Discussion by dkobayas-cyber** [pyvista/pyvista#4749] [Q&A] [answered]
I'd like to print data of a point whenever it is left-cliked by mouse. It is similar to the [previous post](https://github.com/pyvista/pyvista/discussions/4628#discussion-5372412) , but instead of printing the data in the output terminal, I want to show it inside of the plot. Below is a quick example: 

```python

# Point set 
N = 10
keypoints = mesh.points[:N]
pset = pv.PointSet(keypoints)
pset.point_data['Kp_idx'] = range(N)

# Callback function to show an index of a picked point
def my_callback(_point, picker):
    # picker is a vtkPointPicker
    idx = picker.GetPointId()
    if idx < N:
        point_idx = pset.point_data['Kp_idx'][idx]
        # Print the index of the picked point <-- Need to modify
        print(f'Picked kp index: {point_idx}\n', end="")

# Plot 
pl = pv.Plotter(off_screen=True)

# Plot point set 
pl.add_mesh(pset, color='red', point_size=12, 
            render_points_as_spheres=True, opacity=0.5)
# When picked, shows the index 
pl.enable_point_picking(callback=my_callback, color='blue',
                        use_picker=True, left_clicking=True, 
                        show_message='Want to show the output here!')
# Mesh of bunny
pl.add_mesh(mesh, opacity=0.5)
pl.show()
```
![image](https://github.com/pyvista/pyvista/assets/85417498/d0d8a9db-14d5-48a9-9c0e-5006b2fa0bcf)

I'd like to know how to modify this code so that 
* When a red point is picked, `point_idx` value is displayed on the upper-left corner of the image.
* If another red point is picked, the previous message is overwritten by a new `point_idx` value.

**Comment by darikg** [comment, marked as answer]
`enable_point_picking` stores a reference to the `vtkCornerAnnotation` message in the plotter's `_picking_text` variable, which you could access in the callback, or instead create your own corner annotation with the `add_text` method. Updating the text is slightly awkward because the nice pyvista wrapping around the corner enum is hidding inside the `add_text` method. but this works:

```python
from vtkmodules.vtkRenderingAnnotation import vtkCornerAnnotation
...

pl = pv.Plotter()
picking_text = pl.add_text('placeholder', position='upper_left')  # vtkCornerAnnotation

def my_callback(_point, picker):
    point_idx =  ...
    picking_text.SetText(vtkCornerAnnotation.UpperLeft, f'Picked kp index: {point_idx}')

pl.enable_point_picking(show_message=False, ...)  # since we made one manually
```

**Reply by dkobayas-cyber** [reply]
@darikg It does exactly what I wanted! Thank you for your help.

**Comment by dkobayas-cyber** [comment]
@darikg I have a follow-up question. When we have two models, is it possible to enable the point-picking for just one of the models? In the example above, I want to allow the point picking only for the `pset` (point set) but not for the `mesh` (bunny).

**Reply by darikg** [reply]
I think it should just be `plotter.add_mesh(bunny, pickable=False)`
